subset_vds.py

```diff
@@ -0,0 +1,373 @@
+        )
+    subset_sample_list = sample(starting_samples, n_samples)
+    return subset_sample_list
+
+
+def subset_by_samples(
+    input_vds: VariantDataset, subset_sample_list: list[str]
+) -> VariantDataset:
+    """subset a vds by samples
+
+    Args:
+        input_vds (VariantDataset): the input vds to filter on
+        subset_sample_list (list[str]): the list of sampels to retain in the vds
+
+    Returns:
+        VariantDataset: the subset vds
+    """
+    subset_vds: VariantDataset
+    subset_vds = filter_samples(input_vds, subset_sample_list)
+    return subset_vds
+
+
+def subset_by_locus(
+    parsed_locus: list[IntervalExpression], input_vds: VariantDataset
+) -> VariantDataset:
+    """subset a vds by locus
+
+    Args:
+        parsed_locus (list[IntervalExpression]): a list of locuses to filter the vds down to
+        input_vds (VariantDataset): the input vds to filter
+
+    Raises:
+        ValueError: throws an error if all rows are removed by filtering
+
+    Returns:
+        VariantDataset: the subset vds
+    """
+    subset_vds: VariantDataset
+    subset_vds = filter_intervals(input_vds, parsed_locus)
+    if subset_vds.variant_data.count_rows() == 0:
+        raise ValueError(
+            f"No rows remain after applying the following locus filters: {parsed_locus}"
+        )
+    return subset_vds
+
+
+def write_outputs(
+    output_formats: list[str],
+    subset_vds: VariantDataset | None,
+    subset_sample_list: list[str] | None,
+    infile_name: str,
+    is_test: bool,
+) -> None:
+    """Writes the vds out in the specified formats
+
+    Args:
+        output_formats (list[str]): list of output formats to write
+        subset_vds (VariantDataset | None): the resulting vds after subsetting by samples and / or intervals
+        subset_sample_list (list[str] | None): the samples left in the output data, when subsetting by samples
+        infile_name (str): the name of the input vds, to append '_subset' to
+        is_test (bool): controls whether to write to test or not
+
+    Raises:
+        FileExistsError: throws an error if any of the proposed output paths exist, as it will not overwrite them
+    """
+    if "vds" in output_formats:
+        if to_path(output_path(f"{infile_name}_subset")).exists():
+            raise FileExistsError(
+                f'The file {to_path(output_path("{infile_name}_subset"))} exists. Refusing to overwrite it.'
+            )
+        subset_vds.write(output_path(f"{infile_name}_subset", test=is_test))
+
+    subset_dense_mt: MatrixTable | Table | Any = to_dense_mt(subset_vds)
+    subset_dense_mt = split_multi_hts(subset_dense_mt)
+
+    for format in output_formats:
+        if format == "bed":
+            export_plink(
+                subset_dense_mt,
+                output_path(f"{infile_name}_subset", test=is_test),
```

Comment:
`output_path` will handle this automatically - output_path builds an output location based on a few things in config:
1. default bucket for the dataset
2. output_prefix
3. file path/suffix

[output_path](https://github.com/populationgenomics/cpg-utils/blob/main/cpg_utils/config.py#L421) calls [dataset_path](https://github.com/populationgenomics/cpg-utils/blob/main/cpg_utils/config.py#L297), which builds the full path.

For a test run, the only paths in config are to test buckets, so it will only know how to build test output paths.

For main/full runs, there will be both test and main bucket paths in config. This `test` toggle is designed so that on a full/main permission run you can choose to direct the output to a test bucket anyway.

From the way you're using this (based on the access level), you'd be directing the output to the location it would be writing to by default, so the test argument doesn't really help you here.

Reply:
Makes sense, I can remove this